### PR TITLE
[Document Provider] Account management

### PR DESCRIPTION
**Pull request by abelgardep**
Implements #2518 

____

Test plan: https://github.com/owncloud/QA/blob/master/Mobile/Android/Release_2.11.0/2528-document_provider_account_manager.md

bugs & improvements: 

- [x] (1) https://github.com/owncloud/android/pull/2528#issuecomment-489969621 [FIXED]
- [x] (2) https://github.com/owncloud/android/pull/2528#issuecomment-492258721 [NOT REPRODUCIBLE]

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/ui/dialog/RemoveAccountDialogFragment.java [hunk 1]
```diff
@@ -103,6 +107,11 @@ public void onConfirmation(String callerTag) {
         AccountManager am = AccountManager.get(parentActivity);
         AccountManagerCallback<Boolean> callback = (AccountManagerCallback<Boolean>) parentActivity;
         am.removeAccount(mTargetAccount, callback, new Handler());
+
+        // Notify removal to Document Provider
+        String authority = getResources().getString(R.string.document_provider_authority);
+        Uri rootsUri = DocumentsContract.buildRootsUri(authority);
+        Objects.requireNonNull(getContext()).getContentResolver().notifyChange(rootsUri, null);
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/ui/dialog/RemoveAccountDialogFragment.java]: Just a question, why do you use `Objects` and not something like:

```
if (getContext() != null) {
    getContext().getContentResolver().notifyChange(rootsUri, null);
}
```

**Comment by jesmrec** [comment]
### (1)

1. Add an account
2. In oC, open drawer -> manage accounts
3. Add account from account manager
4. Add correct credentials of a new account -> account added and account manager displayed with both accounts
5. Minimize oC app and open Downloads/Files app, and select the account added in step 3

Current: app crashes (maybe because file list was not fetched yet)
Expected: not crash. To display, here depends on what can be showed at such point.

Nexus 5X v8

**Comment by davigonz** [comment]
@jesmrec https://github.com/owncloud/android/pull/2528#issuecomment-489969621 is ready to test

**Comment by jesmrec** [comment]
### (2)

By repeating the steps in (1), the app does not crash anymore. But, even browsing in the app through the account, in "Downloads" is shown `Can't load content at the moment`

**Comment by jesmrec** [comment]
The issue explained in (2) is not always reproducible by using the same steps. Let's give a try to the current implementation and in case of someone or somewhere can get more specific steps, we will open a new issue.

Approved.

**Pull request merged**